docs/references/api/resource_embedding.rst

```diff
@@ -1189,6 +1199,69 @@ You can use this to get the columns of a join table in a many-to-many relationsh
      }
    ]
 
+One-to-many and Many-to-many relationships
+------------------------------------------
+
+For these relationships, the spread embedded resource columns will be aggregated into arrays.
+
 .. note::
+  This is disabled by default, since aggregates can cause performance issues.
+  It can be activated by setting :ref:`db-aggregates-enabled` to ``true``.
 
-  The spread operator ``...`` is borrowed from the Javascript `spread syntax <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax>`_.
+.. code-block:: bash
+
+   # curl -g "http://localhost:3000/directors?select=first_name,...films(film_titles:title,film_years:year)&first_name=like.Quentin*"
+
+  curl --get "http://localhost:3000/directors" \
+    -d "select=first_name,...films(film_titles:title,film_years:year)" \
+    -d "first_name=like.Quentin*"
+
+.. code-block:: json
+
+   [
+     {
+       "first_name": "Quentin",
+       "film_titles": [
+         "Pulp Fiction",
+         "Reservoir Dogs"
+       ],
+       "film_years": [
+         1994,
+         1992
+       ]
+     }
+   ]
+
+The order of the values inside the resulting array is unspecified.
+However, if more than one embedded column is selected, `it is safe to assume <https://www.postgresql.org/message-id/15950.1491843689%40sss.pgh.pa.us>`_ that all of them will return the values in the same unspecified order.
+From the previous example, we can say that "Pulp Fiction" was made in 1994 and "Reservoir Dogs" in 1992.
+
+It is expected to get duplicated or null values inside the aggregated array.
+So you can use the ``..`` operator to solve this.
+For example, a film could get nominated in the same year for different competitions.
```

Comment:
So far I'm against adding this syntax. This case is too ad hoc, a distinct only for the spread operator. It's not intuitive at all and it's more cognitive load for the user.

I keep getting asked about `distinct` support so we might be able to offer it in a more composable way.

Maybe there's a way to avoid this here in this PR, I'll keep reviewing.

Reply:
I understand the use case, but why force this behavior on the spread operator? Couldn't this work by using [Stripped Nulls](https://docs.postgrest.org/en/v12/references/api/resource_representation.html#stripped-nulls) on the regular spread `...` instead?

```
$ curl --get "http://localhost:3000/films" \
    -H "Accept: application/vnd.pgrst.array+json;nulls=stripped" \
    -d "select=title,..competitions(competition_years:year)"
```

Reply:
For `distinct`, wouldn't be better to add support for it as an aggregate function? Like:

```http
GET /films?select=title,..competitions(competition_years:year.distinct())
```

Reply:
> Couldn't this work by using [Stripped Nulls](https://docs.postgrest.org/en/v12/references/api/resource_representation.html#stripped-nulls) on the regular spread `...` instead?

That's a nice alternative. Although I'm not entirely sure since `nulls=stripped` is translated to `jsonb_strip_nulls()` directly, which the PostgreSQL docs define as:

> Deletes all object fields that have null values from the given JSON value, recursively. Null values that are not object fields are untouched.
>
> `json_strip_nulls('[{"f1":1, "f2":null}, 2, null, 3]') → [{"f1":1},2,null,3]`

Still, we could define it as removing `null`s from the to-many spread only. But it won't include cases where the user wanted no nulls in the object fields, but respect the nulls in the to-many arrays (maybe an edge case?).

> For `distinct`, wouldn't be better to add support for it as an aggregate function?

I like this idea. Do you think it's better to implement `DISTINCT ON` for normal columns too? Otherwise it'd only work for to-many spreads (as `array_agg(DISTINCT ...)`) and would need to throw an error when it doesn't comply.

Reply:
> I like this idea. Do you think it's better to implement DISTINCT ON for normal columns too? Otherwise it'd only work for to-many spreads (as array_agg(DISTINCT ...)) and would need to throw an error when it doesn't comply.

Yeah, that sounds good. It's definitely a separate issue/pr though. We should first merge the regular to-many spread here, which is already useful.